mathics/builtin/comparison.py

```diff
@@ -333,49 +345,12 @@ def apply(self, items, evaluation):
         if not all(val == SymbolTrue for val in is_exact_vals):
             return self.apply_other(items, evaluation)
         args = self.numerify_args(items, evaluation)
-        wanted = operators[self.get_name()]
-        for x, y in itertools.combinations(args, 2):
-            if isinstance(y, Complex):
-                x, y = y, x
-            if isinstance(x, Complex):
-                if isinstance(y, Complex):
-                    c = do_cmp(x.real, y.real)
-                    if c is None:
-                        return
-                    if c not in wanted:
-                        return SymbolFalse
-                    c = do_cmp(x.imag, y.imag)
-                    if c is None:
-                        return
-                    if c not in wanted:
-                        return SymbolFalse
-                    else:
-                        return SymbolTrue
-                else:
-                    c = do_cmp(x.imag, Integer(0))
-                    if c is None:
-                        return
-                    if c not in wanted:
-                        return SymbolFalse
-                    c = do_cmp(x.real, y.real)
-                    if c is None:
-                        return
-                    if c not in wanted:
-                        return SymbolFalse
-                    else:
-                        return SymbolTrue
-            # if isinstance(x, String) or isinstance(y, String):
-            # if not (isinstance(x, String) and isinstance(y, String)):
-            #    c = 1
-            # else:
-            #    c = cmp(x.get_string_value(), y.get_string_value())
-            else:
-                c = do_cmp(x, y)
+        for x, y in self.get_pairs(args):
+            c = do_cmp_cplx(x, y)
```

Comment:
Even though `do_cmp_cplx` calls `do_cmp()` this is again an equal function not a `cmp` function as far as I can tell.